### ISSUE TITLE
GET /bellschedules fails when connection to auth0 management API times out

### ISSUE THREAD

**Issue by MoralCode** [MoralCode/ClassClockAPI#10]
This results in the Admin panel not showing any data.

This is likely due to the api trying to hold a single connection to the management API (which is used for checking roles and other access things) instead of using a connection pool or some other way of doing it.

That said this can also be reproduced by omitting the auth0 client ID and or secret environment variables

**Comment by MoralCode** [comment]
from the requests docs:
> Requests allows you to send HTTP/1.1 requests extremely easily. There’s no need to manually add query strings to your URLs, or to form-encode your POST data. Keep-alive and HTTP connection pooling are 100% automatic, thanks to [urllib3](https://github.com/urllib3/urllib3).

https://requests.readthedocs.io/en/latest/ 

so maybe this isnt the problem...

**Comment by MoralCode** [comment]
This still seems to be a problem, except its now manifesting as a 500 server error due tot he following stack trace:

```
[2022-11-20 00:11:15 +0000] [9] [ERROR] Error handling request /v0/bellschedules
Traceback (most recent call last):
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 176, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/root/.local/share/virtualenvs/classclock-api-8K7EYNFV/lib/python3.8/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/classclock-api/common/helpers.py", line 433, in wrapper
    return func(*args, **kwargs)
  File "/classclock-api/common/helpers.py", line 378, in decorated
    return func(*args, **kwargs)
  File "/classclock-api/common/helpers.py", line 396, in decorated
    is_admin = check_for_role("admin")
  File "/classclock-api/common/helpers.py", line 292, in check_for_role
    role_names = [r["name"].lower() for r in roles_json]
  File "/classclock-api/common/helpers.py", line 292, in <listcomp>
    role_names = [r["name"].lower() for r in roles_json]
TypeError: string indices must be integers
```

The problem does not manifest if the api was recently started (so restarting is a valid workaround).

what i suspect happens is:


1. api starts running
2. some (long) time goes by, the server loses its active connection to the auth0 management API
3. an admin tries to use the admin portal or any other protected endpoint
4. the app tries to validate that they have permission for that endpoint
5. in doing so the app tries to make a call to the auth0 management API
6. instead of the normal JSON response (`[{'id': 'rol_whateverID', 'name': 'name', 'description': 'A person with permissions'}]`) the app is receiving something that it thinks is a string.
7. it tries to check the string as if it were a list of roles and crashes with the above stack trace.

Adding better role checking did not solve the problem

**Comment by MoralCode** [comment]
This error also causes the client to see a 500 Server error and be unable to process CORS headers, leading to more errors

**Comment by MoralCode** [comment]
This token is apparrently of the form
```
{
  "access_token":"eyJz93a...k4laUWw",
  "token_type":"Bearer",
  "expires_in":86400
}
```
and the expiry seems to be about a day which makes sense

**Comment by MoralCode** [comment]
should be fixed in fc9dee436a52e624d2ce8d049cb8390eacc4119d
